modules/tensor_mechanics/include/materials/GeneralizedMaxwellModel.h

```diff
@@ -0,0 +1,49 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef GENERALIZEDMAXWELLMODEL_H
+#define GENERALIZEDMAXWELLMODEL_H
+
+#include "GeneralizedMaxwellBase.h"
+
+class GeneralizedMaxwellModel;
+
+template <>
+InputParameters validParams<GeneralizedMaxwellModel>();
+
+/**
+ * This class is an implementation of a generalized Maxwell model
+ * with constant mechanical properties. It consists of an arbitrary number
+ * of Kelvin-Voigt units associated in series with a single spring, and an
+ * optional dashpot. Having an optional dashpot may lead to divergence of
+ * the time-stepping scheme and should be used with care.
+ */
+class GeneralizedMaxwellModel : public GeneralizedMaxwellBase
+{
+public:
+  GeneralizedMaxwellModel(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpViscoelasticProperties();
+  virtual void computeQpViscoelasticPropertiesInv();
+
+  /**
+   * The elasticity tensor associated with the first spring. This is
+   * !not! the true elasticity tensor of the material
+   */
+  RankFourTensor _C0;
```

Comment:
Any reason to keep this separate from the other ones in _Ci?

Reply:
Yes, this is a separate spring that is not associated with any dashpot, while the _Ci and the _eta_i are paired together. It is involved in the computation of the total stiffness of the material, but not of the creep strain itself.